src/test/ui/error-codes/E0121.stderr

```diff
@@ -2,7 +2,10 @@ error[E0121]: the type placeholder `_` is not allowed within types on item signa
   --> $DIR/E0121.rs:1:13
    |
 LL | fn foo() -> _ { 5 }
-   |             ^ not allowed in type signatures
+   |             ^
+   |             |
+   |             not allowed in type signatures
+   |             help: replace `_` with the correct return type: `i32`
 
 error[E0121]: the type placeholder `_` is not allowed within types on item signatures
```

Comment:
Heh, you could also implement it for `const`/`static` with type `_`.

Reply:
We could also add parser recovery for when no type is provided syntactically, for example: `const FOO = 42;`. I think more people would attempt that

Reply:
Help for `const FOO = 42;` looks nice, that's how I usually forget about it in `const`'s, though I'm not sure what'd be the correct parser recovery, replacing the type with `Infer` and then failing here with proper help?

Reply:
Yes, basically delegate to the AST representation of `const $name: _ = $expr;` and make sure to error in the parser.

Reply:
@Centril But would we have the correct type to suggest in the parser? (I assume you implied the type of `self.parse_expr()` on the right side of const declaration)

Reply:
The correct type in the parser would be `Infer`.

Reply:
But then the suggestion for `const FOO = 42` would be the same as for `const FOO: _ = 42` - to *replace* the `_` even though the user hasn't written it, could we perhaps denote some sort of type-absence to handle that propely?

Reply:
We can skip this for now, tbh.

Reply:
Yeah, let's follow up later.

@lundibundi We should take care not to suggest replacing `_` with something since the user did not write that. Instead we should indicate that the type is missing and that we think the type is so and so.